apps/pie-storybook/stories/pie-radio.stories.ts

```diff
@@ -88,16 +88,75 @@ export default radioStoryMeta;
 
 const Template = ({
     checked, disabled, defaultChecked, name, required, slot, value,
-}: RadioProps) => html`
-    <pie-radio
-        ?checked="${checked}"
-        ?disabled="${disabled}"
-        ?defaultChecked="${defaultChecked}"
-        ?required="${required}"
-        name="${ifDefined(name)}"
-        .value="${value}">
-        ${sanitizeAndRenderHTML(slot)}
-    </pie-radio>
-`;
+}: RadioProps) => {
+    const [, updateArgs] = UseArgs();
+
+    const onChange = (event: InputEvent) => {
+        const radioElement = event.target as HTMLInputElement;
+        updateArgs({ checked: radioElement.checked });
+    };
+
+    return html`
+        <pie-radio
+            ?checked="${checked}"
+            ?disabled="${disabled}"
+            ?defaultChecked="${defaultChecked}"
+            ?required="${required}"
+            name="${ifDefined(name)}"
+            .value="${value}"
+            @change="${onChange}">
+            ${sanitizeAndRenderHTML(slot)}
+        </pie-radio>`;
+};
+
+const ExampleFormTemplate: TemplateFunction<RadioProps> = ({
+    value,
+    name,
+    checked,
+    defaultChecked,
+    disabled,
+    required,
+    slot,
+}: RadioProps) => {
+    const [, updateArgs] = UseArgs();
+
+    const onChange = (event: InputEvent) => {
+        const radioElement = event.target as HTMLInputElement;
+        updateArgs({ checked: radioElement.checked });
+    };
+
+    return html`
+        <form id="testForm">
+            <pie-radio
+                .value="${value}"
+                name="${ifDefined(name)}"
+                ?checked="${checked}"
+                ?defaultChecked="${defaultChecked}"
+                ?disabled="${disabled}"
+                ?required="${required}"
+                @change="${onChange}">
+                ${sanitizeAndRenderHTML(slot)}
```

Comment:
I see that we don't use the sanitize function on every component with slotted content, is there a rule to determine when to know when use that or is it the case for wrapping the remaining ones in a follow-up?

Reply:
Good question. It seems to allow you to pass HTML-like text into the slot and it will be rendered as HTML, so I think we should be using that whenever we want the consumer to be able to do the same (which will basically be everywhere, otherwise we may as well just use a string prop instead of a slot!).

| `${sanitizeandRenderHTML(slot)}` | `${slot}` |
| --- | --- |
| ![2024-09-24 13_25_08](https://github.com/user-attachments/assets/7ed88acc-cbe9-4142-8402-5f900af3b0a6) | ![2024-09-24 13_24_43](https://github.com/user-attachments/assets/6ed47ae5-4656-4c6a-b7ab-d9dbaf8ee2fc) |

Reply:
It makes all the sense! Thanks for the detailed answer!